src/smt/optimization_solver.cpp

```diff
@@ -16,78 +16,58 @@
 #include "smt/optimization_solver.h"
 
 #include "omt/omt_optimizer.h"
+#include "smt/assertions.h"
 
 using namespace cvc5::theory;
 using namespace cvc5::omt;
 namespace cvc5 {
 namespace smt {
 
-/**
- * d_activatedObjective is initialized to a default objective:
- * default objective constructed with Null Node and OBJECTIVE_UNDEFINED
- *
- * d_savedValue is initialized to a default node (Null Node)
- */
-
-OptimizationSolver::OptimizationSolver(SmtEngine* parent)
-    : d_parent(parent),
-      d_activatedObjective(Node(), ObjectiveType::OBJECTIVE_UNDEFINED),
-      d_savedValue()
+OptimizationResult::ResultType OptimizationSolver::checkOpt()
 {
-}
-
-OptimizationSolver::~OptimizationSolver() {}
-
-OptResult OptimizationSolver::checkOpt()
-{
-  // Make sure that the objective is not the default one
-  Assert(d_activatedObjective.getType() != ObjectiveType::OBJECTIVE_UNDEFINED);
-  Assert(!d_activatedObjective.getNode().isNull());
-
+  Assert(d_objectives.size() > 0);
+  // just a single objective optimization
```

Comment:
From this comment and from the code it seems like only single objective optimization is currently supported. Is that right? If so, please add assertions in the code that will not allow to create more than one objective (e.g., here assert that the objective size is 1).

Reply:
YES. and this is *preliminary* support for multiple objective. 
If this is approved optimization algorithms will be up next.

Reply:
It allows, just picks the first one and optimize with others no effect.

Reply:
and multiobj algorithms will be up right next.

Reply:
Thanks for adding the comment. Please also add an assertion
`Assert(d_objectives.size() == 1);` and add a comment next to it saying we currently process a single objective. When you change the code so that it processes more objectives, simply remove the assertion.

I saw that you mentioned that this is done: https://github.com/CVC4/CVC4/pull/6459#discussion_r624153355

But I can't find the assertion itself.

Reply:
Oh I forgot, sorry. Now it's fixed.